puppet/modules/quickstack/manifests/neutron/plugins/nexus1000v.pp

```diff
@@ -0,0 +1,127 @@
+# policy profiles.
+# Default value: False, indicating all tenants can access all policy profiles.
+# restrict_policy_profiles = False
+
+class quickstack::neutron::plugins::nexus1000v (
+  $default_policy_profile		= 'default-pp',
+  $default_vlan_network_profile		= 'default-vlan-np',
+  $default_vxlan_network_profile	= 'default-vxlan-np',
+  $poll_duration			= '60',
+  $http_pool_size			= '4',
+  $http_timeout				= '15',
+  $n1kv_vsm_ip                          = undef,
+  $n1kv_vsm_password                    = undef,
+  $n1kv_vsm_username                    = undef,
+  $restrict_policy_profiles		= 'False',
+)
+{
+
+  include neutron::params
+  $package_ensure = 'present'
+  $service_ensure = 'running'
+
+  package { 'cisco-n1kv-ml2-driver':
+    require  => Package['openstack-neutron'],
+  }
+
+  if $::neutron::params::ovs_agent_package {
+    Package['neutron-plugin-ovs-agent'] -> Neutron_plugin_ovs<||>
+    package { 'neutron-plugin-ovs-agent':
+      ensure  => $package_ensure,
+      name    => $::neutron::params::ovs_agent_package,
+    }
+  } else {
+    # Some platforms (RedHat) do not provide a separate
+    # neutron plugin ovs agent package. The configuration file for
+    # the ovs agent is provided by the neutron ovs plugin package.
+    Package['neutron-plugin-ovs'] -> Neutron_plugin_ovs<||>
+    Package['neutron-plugin-ovs'] -> Service['ovs-cleanup-service']
+
+    ensure_resource('package', 'neutron-plugin-ovs', {
+      ensure => $package_ensure,
+      name   => $::neutron::params::ovs_server_package,
+    })
+  }
+
+  service { 'neutron-plugin-ovs-service':
+    ensure  => $service_ensure,
+    name    => $::neutron::params::ovs_agent_service,
+    enable  => $enabled,
+    require => Class['neutron'],
+  }
+
+  if $::neutron::params::ovs_cleanup_service {
+    service {'ovs-cleanup-service':
+      name   => $::neutron::params::ovs_cleanup_service,
+      enable => $enabled,
+    }
+  }
+
+  $extension_drivers  = "cisco_n1kv_ext"
+  neutron_plugin_ml2 {
+    'ml2/extension_drivers'                          : value => $extension_drivers;
+    'ml2_cisco_n1kv/default_policy_profile'          : value => $default_policy_profile;
+    'ml2_cisco_n1kv/default_vlan_network_profile'    : value => $default_vlan_network_profile;
+    'ml2_cisco_n1kv/default_vxlan_network_profile'   : value => $default_vxlan_network_profile;
+    'ml2_cisco_n1kv/poll_duration'                   : value => $poll_duration;
+    'ml2_cisco_n1kv/http_pool_size'                  : value => $http_pool_size;
+    'ml2_cisco_n1kv/http_timeout'                    : value => $http_timeout;
+    'ml2_cisco_n1kv/restrict_policy_profiles'        : value => $restrict_policy_profiles;
+    'ml2_cisco_n1kv/n1kv_vsm_ips'                    : value => $n1kv_vsm_ip;
+    'ml2_cisco_n1kv/username'                        : value => $n1kv_vsm_username;
+    'ml2_cisco_n1kv/password'                        : value => $n1kv_vsm_password;
+  }
+}
+
```

Comment:
Git yelled about this extra line at the end of file when I applied the patch.  If you remove the line, that worning shoudl go away